### PR TITLE
WIP: Ctf interface

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/CodeInterfaces/CobraTF/CTFinterface.py [hunk 1]
```diff
@@ -11,13 +11,19 @@
...
 from __future__ import division, print_function, unicode_literals, absolute_import
 import warnings
 warnings.simplefilter('default',DeprecationWarning)
 from ctfdata import ctfdata
 from CodeInterfaceBaseClass import CodeInterfaceBase
+from GenericCodeInterface import GenericParser
+
 import os
-class CobraTF(CodeInterfaceBase):
+class CTF(CodeInterfaceBase):
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/CTFinterface.py]: this changes the way we invoke the code interface from the RAVEN input file... why?

File: framework/CodeInterfaces/CobraTF/CTFinterface.py [hunk 2]
```diff
@@ -32,30 +38,66 @@ def finalizeCodeOutput(self,command,output,workingDir):
...
     outputobj.writeCSV(os.path.join(workingDir,output+'.csv'))
+  def findInps(self,inputFiles):
+      """
+        Locates the input files required by CTF (Cobra-TF) Interface
+        @ In, inputFiles, list, list of Files objects
+        @ Out, inputDict, dict, dictionary containing Scale required input files
+      """
+      inputDict = {}
+      CTF = []
+      vuq_param = []
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/CTFinterface.py]: standards!

File: framework/CodeInterfaces/CobraTF/CTFinterface.py [hunk 3]
```diff
@@ -32,30 +38,66 @@ def finalizeCodeOutput(self,command,output,workingDir):
...
+      vuq_mult = []
+
+      for inputFile in inputFiles:
+          if inputFile.getType().strip().lower() == "ctf":
+              CTF.append(inputFile)
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/CTFinterface.py]: standards...indentation is 2 spaces

File: framework/CodeInterfaces/CobraTF/ctfdata.py [hunk 4]
```diff
@@ -18,24 +18,34 @@ class ctfdata:
     Class that parses CTF output file and reads in (output files type: .ctf.out) and write a csv file
   """
 
-  def __init__(self, filen):
+  # def __init__(self, filen):
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/ctfdata.py]: O_o

File: framework/CodeInterfaces/CobraTF/ctfdata.py [hunk 5]
```diff
@@ -18,24 +18,34 @@ class ctfdata:
...
+
     """
       Constructor
       @ In, filen, string, file name to be parsed
       @ Out, None
     """
+
     # check file existence (1)
-    if "ctf.out" not in filen:
+    if ("ctf.out" not in filen) and  ("deck.out" not in filen):
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/ctfdata.py]: deck.out is an hardcoded name of Cobra?

File: framework/CodeInterfaces/CobraTF/ctfdata.py [hunk 6]
```diff
@@ -18,24 +18,34 @@ class ctfdata:
...
     except:
       raise IOError('input file missing')
+
+
     self.deckEndTimeInfo = self.getTimeDeck(self.lines)
     self.majorData, self.headerName = self.getMajor(self.lines)
 
+
+  def returner(self):
+    return 0
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/ctfdata.py]: ?

File: framework/CodeInterfaces/CobraTF/ctfdata.py [hunk 7]
```diff
@@ -76,81 +86,81 @@ def getMajor(self, lines):
...
-    # 2. create numpy array (based on the output reading)
-    if count2 == 0:  # w/o fuel rod
-      dictArray = np.zeros((int(timestepCount), int(
-          (variablesCount + averageVariablesCount) * channelCount * nodeCount + 1)))
-    else:  # with fuel rod (10)
-      dictArray = np.zeros((int(timestepCount), int(
-            (variablesCount + averageVariablesCount + 10) * channelCount * nodeCount + 1)))
+        nodeCount = int(line.split()[1])
+        #print('Channel Count = ', channelCount)
+        #print('Node Count = ', nodeCount)
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/ctfdata.py]: remove commented code

File: framework/CodeInterfaces/CobraTF/ctfdata.py [hunk 8]
```diff
@@ -481,7 +550,204 @@ def getMajor(self, lines):
...
+                      "Error: Unexpected output file format. Check the oufput file (output type 6, heat slab, external flow).")
+            # at end of rod info, reset outType6 booleans
+            if ax == 2:
+              outType6 = False
+              externalFlow = False
+    #print(dictArray.shape)
     return dictArray, header
+
+#foo = ctfdata('/Users/tezba-mac/projects/newParserTests/fold/deck_inflow_2_in.out')
+#foo
```
Comment (alfoa) [inline framework/CodeInterfaces/CobraTF/ctfdata.py]: remove

File: framework/CodeInterfaces/CobraTF/ctfdata.py (hunk 5 above)
Comment (alptezbasaran) [reply framework/CodeInterfaces/CobraTF/ctfdata.py]: Actually yes, but I removed that part. It is either ‘inputName.ctf.out’ or  ‘deck.out’ 

Currently I removed the deck.out option

File: framework/CodeInterfaces/CobraTF/ctfdata.py (hunk 6 above)
Comment (alptezbasaran) [reply framework/CodeInterfaces/CobraTF/ctfdata.py]: Haha.  Removed

File: framework/CodeInterfaces/CobraTF/ctfdata.py (hunk 7 above)
Comment (alptezbasaran) [reply framework/CodeInterfaces/CobraTF/ctfdata.py]: Done

**Comment by alptezbasaran** [comment]
Also I changed the name of the branch in my fork. After final checks of the test formats I am going to create a new pull request so you can remove this one.

**Comment by alfoa** [comment]
```Also I changed the name of the branch in my fork. After final checks of the test formats I am going to create a new pull request so you can remove this one.```
For NQ1 tracking this approach is not the best. Why did you need another branch?

**Comment by alptezbasaran** [comment]
I changed the name of the branch (from ctfInterface to alptezbasaran/ctfInterface), but apparently pull request is not updated. I think I can restore this branch back and modify it but currently "alptezbasaran/ctfInterface".Tell me what is the best practice and I will do it.

**Comment by wangcj05** [comment]
This will be closed, and a new PR with the updated changed will be opened.

**Pull request closed without merging**